main.cpp

```diff
@@ -419,6 +405,38 @@ IntType::~IntType()
     delete value;
 }
 
+IntType& IntType::pow(int num)
+{
+    powInternal(num);
+    return *this;
+}
+
+IntType& IntType::pow(const IntType& itype)
+{
+    powInternal(static_cast<int>(itype));
+    return *this;
+}
+
+IntType& IntType::pow(const FloatType& ftype)
+{
+    int arg = static_cast<int>(ftype);
+    powInternal(arg);
+    return *this;
+}
+
+IntType& IntType::pow(const DoubleType& dtype)
+{
+    int arg = static_cast<int>(dtype);
+    powInternal(arg);
+    return *this;
+}
```

Comment:
What does `powInternal` return?
Why aren't you returning what that function returns?
You should, doing so will eliminate a few lines of repetitive code.
- [ ] revise